### PR TITLE
Specify the correct CUDA architecture for SYCL

### PR DESCRIPTION
**Pull request by masterleinad**
Also, see https://github.com/intel/llvm/issues/1170.

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -111,7 +111,9 @@ IF (KOKKOS_ENABLE_CUDA_CONSTEXPR)
   ENDIF()
 ENDIF()
 
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+IF(KOKKOS_ENABLE_SYCL)
+  SET(CUDA_ARCH_FLAG -Xsycl-target-backend -cuda-gpu-arch)
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:115]: This makes that code unreadable.  Put it elsewhere in a SYCL specific block with a comment

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: cmake/kokkos_arch.cmake (hunk 1 above)
Comment (masterleinad) [reply cmake/kokkos_arch.cmake:115]: I don't quite agree. This is the only place where we set `CUDA_ARCH_FLAG` so it makes sense to handle it here. Of course, I can just move
```
IF(KOKKOS_ENABLE_SYCL)
  SET(CUDA_ARCH_FLAG -Xsycl-target-backend -cuda-gpu-arch)
ENDIF()
```
directly after the existing `IF`-block and overwrite `CUDA_ARCH_FLAG` but that doesn't look better IMHO.

Comment (dalg24) [reply cmake/kokkos_arch.cmake:115]: Change it and make sure you comment.

Comment (dalg24) [reply cmake/kokkos_arch.cmake:115]: (Or find two reviewers that agree with you and think I am wrong)

**Comment by masterleinad** [comment]
Apparently, this doesn't solve the issue yet but #3741 does. Let's make `wip` for now.

**Comment by masterleinad** [comment]
Linked in the compiler bugs, no reason to keep it open.

**Pull request closed without merging**